### ISSUE TITLE
`reportAssignmentType` false positive class with `Final` attribute conforms to protocol with property

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1635]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&enableExperimentalFeatures=true&reportMissingModuleSource=false&strictGenericNarrowing=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoBiqAhgDYA0UACuDGAMZikBQr9pxAzp4WGABQ0wdRqQCUALmZQZUAAIJwCAKZ4402QBNlwKADcyAV2X9Oy0sDFQAtAD5MKGBKgA6N2w7coAIWIgpsvpGys5EKGQA2qgwALpQALxQAIysAMRQACqIylAARL4guZg8KMJQXJxIaOEARqQ5dFDa7H7KmrDZeQR8uczpMvl%2BRUg8qIzYxDBIdTkA7kgwABZQisIMTF09fYF5BqTGwyVl2sCobeU8xFAAwp6cAGp%2BDiu06yzAfKF8CT5%2B-GJAA)

```python
from typing import Final, Protocol


class Foo(Protocol):
    @property
    def value(self) -> int: ...

class Bar:
    value: Final[int] = 1

# Type "Bar" is not assignable to declared type "Foo"
#   "Bar" is incompatible with protocol "Foo"
#     "value" is not defined as a ClassVar in protocol
foo: Foo = Bar()
```

related: #775
